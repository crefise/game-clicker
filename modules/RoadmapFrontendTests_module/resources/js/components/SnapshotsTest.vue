<script setup lang="ts">
import { defineProps } from 'vue'

const props = defineProps({
  message: {
    type: String,
    required: true,
  },
})
</script>

<template>
  <div style="border: 1px black solid; padding: 10px">
    <div>
      <h4><em>This test is created to show knowledge about snapshots. </em></h4>
    </div>
    <div v-text="props.message"></div>
    <div>
      <pre style="background: #cbd5e0">
        test('Snapshot test', () => {
          const wrapper = shallowMount(SnapshotsTest, {
            props: {
              message: 'Hello new snapshot!',
            }
          })
          expect(wrapper.html()).toMatchSnapshot()
        })
      </pre>
    </div>
  </div>
</template>

<style scoped>

</style>