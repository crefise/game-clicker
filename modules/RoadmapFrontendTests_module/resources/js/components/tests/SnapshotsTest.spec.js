import { shallowMount } from '@vue/test-utils'
import SnapshotsTest from '../SnapshotsTest';

test('Snapshot test', () => {
  const wrapper = shallowMount(SnapshotsTest, {
    props: {
      message: 'Hello new snapshot!',
    }
  })
  expect(wrapper.html()).toMatchSnapshot()
})