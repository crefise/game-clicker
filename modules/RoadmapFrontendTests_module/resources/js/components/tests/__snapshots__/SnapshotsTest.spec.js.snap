// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
"<div style="border: 1px solid black; padding: 10px;">
  <div>
    <h4><em>This test is created to show knowledge about snapshots. </em></h4>
  </div>
  <div>Hello new snapshot!</div>
  <div><pre style="background: rgb(203, 213, 224);">        test('Snapshot test', () =&gt; {
          const wrapper = shallowMount(SnapshotsTest, {
            props: {
              message: 'Hello new snapshot!',
            }
          })
          expect(wrapper.html()).toMatchSnapshot()
        })
      </pre>
  </div>
</div>"
`;
